src/main/java/sentinel/api/GeminiApi.java

```diff
@@ -0,0 +1,101 @@
+package sentinel.api;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The GeminiApi handles all AI related requests.
+ */
+
+public class GeminiApi {
+
+    /**
+     * @param input The prompt sent to Gemini.
+     * @return The text result from Gemini.
+     */
+
+    public static String query(String input) {
```

Comment:
Very interesting approach. Hopefully, it does not take too long to query the API if not the app might seem unresponsive :). 

Also have interested to find out if there is any fall back if user is not connected to the internet?

Reply:
Currently in my latest commit, users without internet or RequestTimeOuts will fallback on the standard LocalDateTime.parse(<String input>) method where users have to abide by the format YYYY-MM-DDTHH:M:S. Thanks for asking!